metal/cloud_test.go

```diff
@@ -211,14 +209,89 @@ func TestHasClusterID(t *testing.T) {
 }
 
 // builds an Equinix Metal client
-func constructClient(authToken string, baseURL *string) *packngo.Client {
-	client := retryablehttp.NewClient()
+func constructClient(authToken string, baseUrl *string) *metal.APIClient {
+	configuration := &metal.Configuration{
+		DefaultHeader:    make(map[string]string),
+		UserAgent:        "metal-go/0.29.0",
+		Debug:            false,
+		Servers:          metal.ServerConfigurations{},
+		OperationServers: map[string]metal.ServerConfigurations{},
+	}
 
-	// client.Transport = logging.NewTransport("EquinixMetal", client.Transport)
-	if baseURL != nil {
-		// really should handle error, but packngo does not distinguish now or handle errors, so ignoring for now
-		client, _ := packngo.NewClientWithBaseURL(ConsumerToken, authToken, client.StandardClient(), *baseURL)
-		return client
+	servers := metal.ServerConfigurations{
+		{
+			URL:         "https://api.equinix.com/metal/v1",
+			Description: "No description provided",
+		},
+	}
+	if baseUrl != nil {
+		servers = metal.ServerConfigurations{
+			{
+				URL:         *baseUrl,
+				Description: "No description provided",
+			},
+		}
 	}
-	return packngo.NewClientWithAuth(ConsumerToken, authToken, client.StandardClient())
+
+	configuration.Servers = servers
+	configuration.AddDefaultHeader("X-Auth-Token", authToken)
+	configuration.UserAgent = fmt.Sprintf("cloud-provider-equinix-metal/%s %s", version.Get(), configuration.UserAgent)
+	return metal.NewAPIClient(configuration)
+}
+
+func (s *MockMetalServer) CreateHandler() http.Handler {
```

Comment:
This is good stuff!  It might be nice to move the `MockMetalServer` definition and this `CreateHandler` function to a separate file so it's easier to generalize & extract in the future; I could imagine the `MockMetalServer` taking a map of route => HTTP method => handler function, but we might need to figure out scope binding so that the handlers functions could reference the mock server's data store.

Reply:
Yes, a store interface can be created which will be internal to testing and inmemory datastore can be implentend for the interface. This will bind which type of store is used for the server.